packages/victory-core/src/victory-util/data.js

```diff
@@ -91,6 +92,64 @@ function cleanData(dataset, props) {
   });
 }
 
+// This method will remove data points that fall outside of the desired domain (non-continuous charts only)
+function formatDataFromDomain(dataset, props) {
+  // domain for checking data points that are outside of the bounds
+  // symbol to determine chart types with a single y value point
+  // interpolation for determining chart types with continuous data points that shouldn't be altered
+  // errorX/Y for determining chart types with error whiskers that shouldn't be altered here
+  const { domain, symbol, interpolation, errorX, errorY } = props;
```

Comment:
I don't think you need to duck-type with `errorX` and `errorY` here, because nulling points with (x, y) values outside the domain is what you end up doing for error bars anyway.